core/src/main/scala/org/genivi/sota/core/campaigns/CampaignLauncher.scala

```diff
@@ -35,8 +45,36 @@ object CampaignLauncher {
     Future.successful(devices.map(_ -> Set(pkg.id)).toMap)
   }
 
-  def launch (deviceRegistry: DeviceRegistry, updateService: UpdateService, id: Campaign.Id, lc: LaunchCampaign)
-             (implicit db: Database, system: ActorSystem, ec: ExecutionContext): Future[List[Uuid]] = {
+  //scalastyle:off parameter.number
+  private def sendMsg(namespace: Namespace, devices: Set[Uuid], pkgId: PackageId, id: Campaign.Id, updateId: Uuid,
+                      messageBus: MessageBusPublisher)
+                     (implicit db: Database, system: ActorSystem, ec: ExecutionContext, mat: ActorMaterializer)
+      : Future[Unit] = {
+    import collection.JavaConverters._
+    implicit val _config = system.settings.config
+    val creds = S3PackageStore.loadCredentials(_config)
+
+    //check that this update involves a treehub image
+    if(pkgId.name.get.startsWith("treehub-") && creds.isDefined) {
+      val s3Store = new S3PackageStore(creds.get)
+      for {
+        pkg      <- db.run(Packages.byId(namespace, pkgId))
+        (uri, _) <- s3Store.retrieve(pkg.id, pkg.uri)
+        file     <- s3Store.retrieveFile(uri)
+        metadata =  Json.fromString(Files.readAllLines(file.toPath).asScala.mkString)
```

Comment:
Hmm this `Json.fromString` makes this into a Json string literal, so it will escape everything. You want to parse this
`io.circe.parser.parse`.